lib/pengine/unpack.c

```diff
@@ -2761,7 +2783,14 @@ unpack_migrate_to_failure(pe_resource_t *rsc, pe_node_t *node, xmlNode *xml_op,
                                       source, TRUE, data_set);
     target_migrate_from_id = pe__call_id(target_migrate_from);
 
-    if ((target_stop == NULL) || (target_stop_id < target_migrate_from_id)) {
+    if (/* If the resource has no lrm_rsc_op history on the target,
+         * that means its state is unknown there and it will be probed.
+         * Don't just consider it running there. We will get back here anyway in
+         * case the probe detects it's running there.
+         */
```

Comment:
Unfortunately, we can't assume there will be a probe. There's resource-discovery (which we don't have easy access to at this point), or something could end up blocking the probe (maintenance mode on relevant node, relevant node is remote node whose connection is blocked, etc.). In that case, the resource could potentially be left multiply active.

However that's not too different from the situation without migration when a resource is already active on one node and a probe is blocked on some other node, so I suppose we can keep this. I can't think of anything better.

Reply:
Indeed. Functionized it now.